### ISSUE TITLE
io: Extra linebreaks in captured output

### ISSUE THREAD

**Issue by jbohren** [jbohren-forks/catkin_tools#4]
@wjwwood https://github.com/catkin/catkin_tools/pull/196#issuecomment-113603700

> I've also noticed that the output from warnings (maybe stderr, maybe all captured output) has extra line breaks in them. For example:

> ```
C
M
ake Warning (dev):
  Policy C
MP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.
  Use the cmake_policy command to
  set the policy and suppress this warning.
  MACOSX_RPATH is not specified
for the following targets:
   dart-core
This warning is for project developers.  Use -Wno-dev to suppress it.
```

> Whereas I would expect it to look like this:

>```
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake --help-policy CMP0042" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.
  MACOSX_RPATH is not specified for the following targets:
   dart-core
This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Comment by jbohren** [comment]
I'm pretty sure this is something that only happens on OS X. I'll add unit tests for the log capturing and output and then we can try to track it down.

**Comment by jbohren** [comment]
Yeah... something is wrong: https://travis-ci.org/jbohren-forks/catkin_tools/jobs/67898420#L2141-L2242

**Comment by jbohren** [comment]
Fixed in e2522af

This was an artifact of the CMake output coloring.

**Comment by wjwwood** [comment]
Cool, nice to see it was fixable and not a product of the asyncio capture.
